File: SDKs/googleapis.com/gkehub/v1/python/src/sdk/models/shared/membershipstate.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class MembershipStateCodeEnum(str, Enum):
    CODE_UNSPECIFIED = "CODE_UNSPECIFIED"
    CREATING = "CREATING"
    READY = "READY"
    DELETING = "DELETING"
    UPDATING = "UPDATING"
    SERVICE_UPDATING = "SERVICE_UPDATING"


@dataclass_json
@dataclass
class MembershipState:
    r"""MembershipState
    MembershipState describes the state of a Membership resource.
    """
    
    code: Optional[MembershipStateCodeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    

File: SDKs/googleapis.com/gkehub/v1alpha2/typescript/src/sdk/models/shared/membershipstate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum MembershipStateCodeEnum {
    CodeUnspecified = "CODE_UNSPECIFIED",
    Creating = "CREATING",
    Ready = "READY",
    Deleting = "DELETING",
    Updating = "UPDATING",
    ServiceUpdating = "SERVICE_UPDATING"
}


// MembershipState
/** 
 * MembershipState describes the state of a Membership resource.
**/
export class MembershipState extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code?: MembershipStateCodeEnum;
}
